src/iuse.cpp

```diff
@@ -1901,63 +1902,111 @@ int iuse::mycus(player *p, item *it, bool t, const tripoint &pos)
-            mon.friendly = -1;
-            if( is_dogfood ) {
+        // This switch handles each petfood for each type of tameable monster.
+        switch( animal_food_type ) {
+        case DOGFOOD:
+            if( mon.type->id == mon_dog_thing ) {
+                p->deal_damage( &mon, bp_hand_r, damage_instance( DT_CUT, rng( 1, 10 ) ) );
+                add_msg( m_bad, _( "You want to feed it the dog food, but it bites your fingers!" ) );
+                if( one_in( 5 ) ) {
+                    add_msg( _( "Apparently it's more interested in your flesh than the dog food in your hand!" ) );
+                    return 1;
+                }
+            } else if( mon.type->id == mon_dog ) {
+                add_msg( m_good, _( "The dog seems to like you!" ) );
+                mon.friendly = -1;
                 mon.add_effect( effect_pet, 1, num_bp, true );
+                return 1;
+            } else {
+                add_msg( _( "There is nothing to be fed here." ) );
+                return 0;
             }
-        } else if( is_dogfood && mon.type->id == mon_dog_thing ) {
-            p->deal_damage( &mon, bp_hand_r, damage_instance( DT_CUT, rng( 1, 10 ) ) );
-            p->add_msg_if_player( m_bad, _( "You want to feed it the dog food, but it bites your fingers!" ) );
-            if( one_in( 5 ) ) {
-                p->add_msg_if_player( _( "Apparently it's more interested in your flesh than the dog food in your hand!" ) );
+
+            break;
+        case CATFOOD:
+            if( mon.type->id == mon_cat ) {
+                add_msg( m_good,
+                         _( "The cat seems to like you!  Or maybe it just tolerates your presence better.  It's hard to tell with cats." ) );
+                mon.friendly = -1;
+                return 1;
+            } else {
+                add_msg( _( "There is nothing to be fed here." ) );
+                return 0;
             }
-        } else {
-            p->add_msg_if_player( _( "The %s seems quite unimpressed!" ), mon.name().c_str() );
-        }
-    } else if( npc * const person_ = g->critter_at<npc>( dirp ) ) {
-        npc &person = *person_;
-        if( query_yn( is_dogfood ?
-            _( "Are you sure you want to feed a person the dog food?" ) :
-            _( "Are you sure you want to feed a person the cat food?" ) ) ) {
-            p->add_msg_if_player( _( "You put your %1$s into %2$s's mouth!" ), it->tname().c_str(), person.name.c_str() );
-            if( person.is_friend() || x_in_y( 9, 10 ) ) {
-                person.say( _( "Okay, but please, don't give me this again. I don't want to eat dog food in the cataclysm all day." ) );
+            break;
+        case CATTLEFODDER:
+            if( mon.has_flag( MF_MILKABLE ) ) {
+                add_msg( m_good, _( "The cow seems to like you! It lets you pat her and seems friendly." ) );
```

Comment:
Switches between "it" and "her", which is awkward, either change "it" to "she" or rearrange the sentence to only have one pronoun.